backend/vagrant.pm

```diff
@@ -0,0 +1,190 @@
+# Copyright © 2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package backend::vagrant;
+
+use strict;
+use warnings;
+
+use base 'backend::virt';
+use bmwqemu;
+use testapi qw(get_required_var get_var);
+
+use Cwd qw(getcwd);
+use File::Temp;
+use File::chdir;
+use IPC::Run;
+
+sub new {
+    my $class = shift;
+    my $self  = $class->SUPER::new;
+    my $vars  = \%bmwqemu::vars;
+    $self->{vagrant_cwd}  = File::Temp->newdir();
+    $self->{box_name}     = get_required_var('VAGRANT_BOX');
+    $self->{boot_timeout} = get_var('VAGRANT_BOOT_TIMEOUT') // 60;
+    $self->{provider}     = get_required_var('VAGRANT_PROVIDER');
+
+    $self->{vagrantfile} = "$self->{vagrant_cwd}/Vagrantfile";
+    bmwqemu::diag("Writing Vagrantfile to $self->{vagrantfile}");
+
+    open(my $fh, '>', "$self->{vagrant_cwd}/Vagrantfile") or die $!;
```

Comment:
It would be better to use `Mojo::File::spurt` here (or actually later when printing) because you want to throw an exception in error cases anyways.